src/main/webapp/app/components/timeSeries/time-series-utils.ts

```diff
@@ -0,0 +1,25 @@
+import { ExtraTimeSeriesEventData, RequiredTimeSeriesEventData } from './TimeSeries';
+
+export function groupTimeSeriesDataByDay(data: RequiredTimeSeriesEventData[] | ExtraTimeSeriesEventData[]) {
+  const dataForEachDay: { [dateString: string]: RequiredTimeSeriesEventData[] & ExtraTimeSeriesEventData[] } = {};
+  for (const eventData of data) {
+    const dateWithoutTime = new Date(eventData.createdAt);
+    dateWithoutTime.setHours(0, 0, 0, 0);
+
+    const dateString = dateWithoutTime.toISOString();
+    if (!dataForEachDay[dateString]) {
+      dataForEachDay[dateString] = [];
+    }
+
+    dataForEachDay[dateString].push(eventData);
+  }
+
+  for (const eventData of Object.values<RequiredTimeSeriesEventData[] | ExtraTimeSeriesEventData[]>(dataForEachDay)) {
+    eventData.sort((event1, event2) => event2.createdAt.getTime() - event1.createdAt.getTime());
+  }
+
+  const days = Object.keys(dataForEachDay);
+  days.sort((day1, day2) => new Date(day2).getTime() - new Date(day1).getTime());
```

Comment:
Is this needed?

Reply:
After getting grouping the timestamps by day, I know that the timestamps within each day are sorted, but there is no guarantee the days themselves are sorted, so I need to sort the days to display in the correct order

Reply:
I'm a bit surprised that if you sort days, the dataForEachDay is sorted as well. I think thats what you trying to do here?

Reply:
I do two sorts. The logic to construct the data structure goes as follows:

1. Create an unsorted mapping of day to all timestamps within that day.
2. For each array of timestamps sort. the timestamps by time.
3. Sort the keys from the map so the days are in order,